torch/distributed/distributed_c10d.py

```diff
@@ -553,6 +553,9 @@ def _new_process_group_helper(world_size,
             _pg_map[pg] = (backend, store)
             _pg_names[pg] = group_name
 
+    if not group_name_:
+        _group_count += 1
```

Comment:
small suggestion: we could simply set `_group_count = len(_pg_names)` all the time here so `_group_count` always reflects the number of pgs we actually have post-creation and we don't drift.

Reply:
This does look better but it changes the behaviour a bit since `_group_count` currently only counts unnamed groups. Is there any risk of breaking something?

Reply:
Actually this would break if process groups are deleted, there could be two groups with the same name. I think `_group_count += 1` without the condition would work though.

Reply:
ah yes, you are correct, `_group_count` tracks pgs created with default names:

https://github.com/pytorch/pytorch/blob/e7e919fc3470b2573afbe9384d24b19b2eadf4c7/torch/distributed/distributed_c10d.py#L169

I don't see any other usages of `_group_count` outside of `distributed_c10d.py` (https://github.com/pytorch/pytorch/search?q=_group_count). I guess what that var really is is an incrementing counter (so no harm in always incrementing it although you might have skip ints in your default pg names)

Based on this I'd say keep the logic as you originally had it. Could you update the comment for `_group_count` to say that its really a "counter" rather than a "count"? 

Apologize for the back-and-forth. Thanks!

Reply:
Done